### PR TITLE
Do not in-place encode arrays in amp-analytics values.

### LINKED ISSUE #2038: AMP-Analytics: Array Encoding Multiple Times
When passing arrays in the "vars" configuration for amp-analytics, the array is first not encoded enough, then encoded too much.

## Example:
```js
'vars': {
  'authors': ["john smith", "jane doe", "last name, first name"]
}
```

### First beacon (visible):
```
authors:john smith,jane doe,last name, first name
```

You can see in the first call, the comma in the last value is not properly encoded which means this will decode as 4 values rather than 3.

### Second beacon:
```
authors:john%20smith,jane%20doe,last%20name%2C%20first%20name
```

This is properly encoded and will decode to three values as expected.

### Third beacon:
```
authors:john%2520smith,jane%2520doe,last%2520name%252C%2520first%2520name
```

Oh noes, that don't look right.

### Fourth beacon:
```
authors:john%252520smith,jane%252520doe,last%252520name%25252C%252520first%252520name
```

Now I'm no pattern matching expert, but this looks like it could be a problem.

### Fifth beacon:
```
authors:john%25252520smith,jane%25252520doe,last%25252520name%2525252C%25252520first%25252520name
```

 (╯°□°）╯︵ ┻━┻

...

### Twenty-sixth beacon:
```
authors:john%252525252525252525252525252525252525252525252520smith,jane%252525252525252525252525252525252525252525252520doe,last%252525252525252525252525252525252525252525252520name%25252525252525252525252525252525252525252525252C%252525252525252525252525252525252525252525252520first%252525252525252525252525252525252525252525252520name
```

:disappointed:

### PR DESCRIPTION
**Pull request by cramforce**
Fixes #2038

### CODE REVIEW

**Comment by jridgewell** [comment]
LGTM.

File: extensions/amp-analytics/0.1/amp-analytics.js [hunk 1]
```diff
@@ -347,14 +347,13 @@ export class AmpAnalytics extends AMP.BaseElement {
    */
   encodeVars_(raw) {
     if (isArray(raw)) {
+      let encoded = [];
```
Comment (jridgewell) [inline extensions/amp-analytics/0.1/amp-analytics.js]: Could be more concise, but this is still LGTM.

```js
if (isArray(raw)) {
  return raw.map(encodeURIComponent).join(',');
}
return encodeURIComponent(raw);
```

Comment (cramforce) [inline extensions/amp-analytics/0.1/amp-analytics.js]: Thats kinda cute :)

Comment (cramforce) [inline extensions/amp-analytics/0.1/amp-analytics.js]: Done

**Pull request merged**